src/utils/helper-functions/utils.ts

```diff
@@ -83,3 +86,48 @@ export function getDecoratorArgument({ expression }: TSESTree.Decorator) {
     ? expression.arguments[0]
     : undefined
 }
+
+export function findNgRxStoreName(
+  context: TSESLint.RuleContext<string, readonly unknown[]>,
+): string {
+  let storeName = ''
+  let i = 0
+  let node: TSESTree.Node
+  const ast = context.getSourceCode().ast
+
+  while (!storeName && (node = ast.body[i++])) {
+    if (!isImportDeclaration(node) || node.source.value !== '@ngrx/store')
+      continue
+
+    const storeImportSpecifier = node.specifiers
+      .map((specifierNode) =>
+        isImportSpecifier(specifierNode) &&
+        specifierNode.imported.name === 'Store'
+          ? specifierNode
+          : undefined,
+      )
+      .find(Boolean)
+
+    if (!storeImportSpecifier) continue
+
+    const [storeImport] = context.getDeclaredVariables(storeImportSpecifier)
+    const storeImportName = storeImport.references
+      .map((ref) => {
+        if (
+          ref.identifier.parent &&
+          isTSTypeReference(ref.identifier.parent) &&
+          ref.identifier.parent.parent &&
+          isTSTypeAnnotation(ref.identifier.parent.parent) &&
+          ref.identifier.parent.parent.parent &&
+          isIdentifier(ref.identifier.parent.parent.parent)
+        ) {
+          return ref.identifier.parent.parent.parent.name
+        }
+        return ''
+      })
+      .find(Boolean)
+    storeName = storeImportName || ''
+  }
+
+  return storeName
+}
```

Comment:
I was thinking we could just use the `findImportDeclarationNode` function present in this file... also, we could change the signature to be  `string | undefined`, instead of returning an empty string.
What do you think?

```suggestion
export function findNgRxStoreName(
  context: TSESLint.RuleContext<string, readonly unknown[]>,
): string | undefined {
  const { ast } = context.getSourceCode()
  const storeImportSpecifier = findImportDeclarationNode(ast, '@ngrx/store')

  if (!storeImportSpecifier) return undefined

  const [{ references }] = context.getDeclaredVariables(storeImportSpecifier)

  return references
    .map(({ identifier: { parent } }) => {
      if (
        parent &&
        isTSTypeReference(parent) &&
        parent.parent &&
        isTSTypeAnnotation(parent.parent) &&
        parent.parent.parent &&
        isIdentifier(parent.parent.parent)
      ) {
        return parent.parent.parent.name
      }
      return undefined
    })
    .find(Boolean)
}
```

Reply:
🚀